client/components/site-selector-modal/index.jsx

```diff
@@ -1,85 +1,89 @@
-	Button = require( 'components/button' ),
-	SitesDropdown = require( 'components/sites-dropdown' ),
-	sitesList = require( 'lib/sites-list' )();
+import Dialog from 'components/dialog';
+import Button from 'components/button';
+import SitesDropdown from 'components/sites-dropdown';
+import sitesListFactory from 'lib/sites-list';
 
-/**
- * Component
- */
-var SiteSelectorModal = React.createClass( {
-	propTypes: {
+const sitesList = sitesListFactory();
+
+class SiteSelectorModal extends Component {
+	static propTypes = {
 		// children: Custom content. Will be displayed above the `SitesDropdown`.
-		children: React.PropTypes.node,
+		children: PropTypes.node,
 		// filter: Function to filter sites to display
-		filter: React.PropTypes.func,
+		filter: PropTypes.func,
 		// hide: Will be called when clicking either button. Should toggle the `isVisible` prop.
-		hide: React.PropTypes.func.isRequired,
+		hide: PropTypes.func.isRequired,
 		// isVisible: Determines if `SiteSelectorModal` will be displayed.
-		isVisible: React.PropTypes.bool.isRequired,
+		isVisible: PropTypes.bool.isRequired,
 		// mainAction: Will be run upon clicking the call-for-action button. Receives `site` as argument.
-		mainAction: React.PropTypes.func.isRequired,
+		mainAction: PropTypes.func.isRequired,
 		// getMainUrl: Use if the call-for-action button should be turned into an `<a>` link. Receives `site` as argument, returns a URL.
-		getMainUrl: React.PropTypes.func,
+		getMainUrl: PropTypes.func,
 		// mainActionLabel: Label for the call-for-action button.
-		mainActionLabel: React.PropTypes.string.isRequired,
+		mainActionLabel: PropTypes.string.isRequired,
 		// className: class name(s) to be added to the Dialog
-		className: React.PropTypes.string
-	},
+		className: PropTypes.string,
+		// from localize()
+		translate: PropTypes.func.isRequired
+	}
+
+	constructor( props ) {
+		super( props );
 
-	getInitialState: function() {
 		const primarySite = sitesList.getPrimary();
 		let filteredSites = sitesList.getVisible();
 
-		if ( this.props.filter ) {
-			filteredSites = filteredSites.filter( this.props.filter );
+		if ( props.filter ) {
+			filteredSites = filteredSites.filter( props.filter );
 		}
 
-		return {
+		this.state = {
 			site: includes( filteredSites, primarySite )
 				? primarySite
-				: filteredSites[0]
+				: filteredSites[ 0 ]
 		};
-	},
+	}
 
-	setSite: function( slug ) {
-		var site = sitesList.getSite( slug );
+	setSite = ( slug ) => {
+		const site = sitesList.getSite( slug );
 		this.setState( { site: site } );
-	},
+	}
 
-	onClose: function( action ) {
+	onClose = ( action ) => {
 		if ( 'mainAction' === action ) {
 			this.props.mainAction( this.state.site );
 		}
 
 		this.props.hide();
-	},
+	}
 
-	onButtonClick: function() {
+	onButtonClick = () => {
 		this.props.mainAction( this.state.site );
-	},
+	}
 
-	getMainLink: function() {
-		var url = this.props.getMainUrl && this.props.getMainUrl( this.state.site );
+	getMainLink = () => {
```

Comment:
Unnecessary binding, as the method is always called with the appropriate context (`this.getMainLink()`).